static/app/views/settings/dynamicSampling/utils/scaleSampleRates.tsx

```diff
@@ -0,0 +1,75 @@
+interface ScalingItem {
+  count: number;
+  sampleRate: number;
+}
+
+/**
+ * Scales the sample rates of items proportionally to their current sample rate.
+ */
+export function scaleSampleRates<T extends ScalingItem>({
+  items,
+  sampleRate,
+}: {
+  items: T[];
+  sampleRate: number;
+}): {
+  scaledItems: T[];
+} {
+  const totalSpans = items.reduce((acc, item) => acc + item.count, 0);
+  const oldSampleRate = items.reduce(
+    (acc, item) => acc + item.sampleRate * (item.count / totalSpans),
+    0
+  );
+
+  if (sampleRate === oldSampleRate) {
+    return {scaledItems: items};
+  }
+
+  if (
+    oldSampleRate === 0 ||
+    oldSampleRate === 1 ||
+    sampleRate === 0 ||
+    sampleRate === 1
+  ) {
+    return {
+      scaledItems: items.map(item => ({
+        ...item,
+        sampleRate,
+      })),
+    };
+  }
```

Comment:
I added special handling for those edge cases as floating point errors caused the algorithm to not alway put every project to exactly 100% when the org rate was set to 100%